pkg/imgpkg/bundle/contents_test.go

```diff
@@ -0,0 +1,156 @@
+	defer os.RemoveAll(bundleDir)
+	bundleImg := &fake.FakeImage{}
+	cfgFile := &v1.ConfigFile{
+		Config: v1.Config{
+			Labels: map[string]string{"dev.carvel.imgpkg.bundle": "true"},
+		},
+	}
+	bundleImg.ConfigFileReturns(cfgFile, nil)
+	fakeRegistry.ImageReturns(bundleImg, nil)
+
+	t.Run("when allowInnerBundles is false, push fails", func(t *testing.T) {
+		subject := bundle.NewContents([]string{bundleDir}, nil, false)
+		imgTag, err := name.NewTag("my.registry.io/new-bundle:tag")
+		if err != nil {
+			t.Fatalf("failed to read tag: %s", err)
+		}
+
+		_, err = subject.Push(imgTag, fakeRegistry, fakeUI)
+		if err == nil {
+			t.Fatalf("expected test to fail, but it did not")
+		}
+		expectedError := "Expected image lock to not contain bundle reference: 'my.registry.io/bundle@sha256:703218c0465075f4425e58fac086e09e1de5c340b12976ab9eb8ad26615c3715'"
+		if err.Error() != expectedError {
+			t.Fatalf("Error message not expected\nExpected: %s\nGot: %s", expectedError, err.Error())
+		}
+	})
+
+	t.Run("when allowInnerBundles is true, push is successful", func(t *testing.T) {
+		subject := bundle.NewContents([]string{bundleDir}, nil, true)
+		imgTag, err := name.NewTag("my.registry.io/new-bundle:tag")
+		if err != nil {
+			t.Fatalf("failed to read tag: %s", err)
+		}
+
+		_, err = subject.Push(imgTag, fakeRegistry, fakeUI)
+		if err != nil {
+			t.Fatalf("not expecting push to fail: %s", err)
+		}
+	})
+}
+
+func TestNewContentsBundleWithImages(t *testing.T) {
+	imagesLockYAML := `---
+apiVersion: imgpkg.carvel.dev/v1alpha1
+kind: ImagesLock
+images:
+- image: my.registry.io/image1@sha256:703218c0465075f4425e58fac086e09e1de5c340b12976ab9eb8ad26615c3715
+`
+	fakeUI := &bundlefakes.FakeUI{}
+	fakeRegistry := &bundlefakes.FakeImagesMetadataWriter{}
+	bundleDir := createBundleDir(t, "inner-bundle", imagesLockYAML)
+	defer os.RemoveAll(bundleDir)
+	bundleImg := &fake.FakeImage{}
+	cfgFile := &v1.ConfigFile{
+		Config: v1.Config{},
+	}
+	bundleImg.ConfigFileReturns(cfgFile, nil)
+	fakeRegistry.ImageReturns(bundleImg, nil)
+
+	t.Run("when allowInnerBundles is false, push is successful", func(t *testing.T) {
+		subject := bundle.NewContents([]string{bundleDir}, nil, false)
+		imgTag, err := name.NewTag("my.registry.io/new-bundle:tag")
+		if err != nil {
+			t.Fatalf("failed to read tag: %s", err)
+		}
+
+		_, err = subject.Push(imgTag, fakeRegistry, fakeUI)
+		if err != nil {
+			t.Fatalf("not expecting push to fail: %s", err)
+		}
+	})
+
+	t.Run("when allowInnerBundles is true, push is successful", func(t *testing.T) {
+		subject := bundle.NewContents([]string{bundleDir}, nil, true)
+		imgTag, err := name.NewTag("my.registry.io/new-bundle:tag")
+		if err != nil {
+			t.Fatalf("failed to read tag: %s", err)
+		}
+
+		_, err = subject.Push(imgTag, fakeRegistry, fakeUI)
+		if err != nil {
+			t.Fatalf("not expecting push to fail: %s", err)
+		}
+	})
+}
+
+func createTempFolder(t *testing.T, prefix string) string {
```

Comment:
any reason why not to use https://golang.org/pkg/io/ioutil/#TempDir?

Reply:
No reason, just copied this function from the e2e package as is. When we consolidate these functions I think we can address this change.

Reply:
> just copied this function from the e2e package as is

that sounds like taking on unnecessary debt. this one should be fixed in this PR.

Reply:
done

Reply:
sorry, what i meant was you can use ioutil.TempDir within createBundleDir instead of having a dedicated method createTempFolder.